### PR TITLE
Writing Custom Code

### PR DESCRIPTION
**Pull request by PoojaB26**
### Description
Provide a brief overview of what this documentation update is about. Explain what sections or topics are being added or revised.

Linear ticket and [magic word](https://linear.app/docs/github#link-prs) 
Fixes DEVR-333
Fixes DEVR-354
Fixes DEVR-355
Fixes DEVR-343

## Type of change
- [ ] Typo fix 
- [x] New feature 
- [x] Enhancement to current docs
- [ ] Removed outdated references
- [ ] Update assets

### CODE REVIEW

File: docs/ff-concepts/adding-customization/cloud-functions.md [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+        return filteredpersons.map(filteredpersons => {
+          return {
+            "name": filteredpersons.name,
+            "age": filteredpersons.age
+          };
+        });
+```
+    
+#### To deploy
+1. Click the `[**</>**]` icon to view the boilerplate code; a popup will open with the updated
```
Comment (pinkeshmars) [inline docs/ff-concepts/adding-customization/cloud-functions.md:122]:
```suggestion
1. Click the `[</>]` icon to view the boilerplate code; a popup will open with the updated 
```

**Review by pinkeshmars — CHANGES_REQUESTED** [review]
@PoojaB26 Added review comments.

File: docs/ff-concepts/adding-customization/custom-widgets.md [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+this example, we will create a `ProductRatingBar` widget that uses a pub.dev dependency to display the rating bar UI. It will also take a callback action to provide the rating value back to the caller.
+
+<div style={{
+    position: 'relative',
+    paddingBottom: 'calc(56.67989417989418% + 41px)', // Keeps the aspect ratio and additional padding
+    height: 0,
+    width: '100%'
+}}>
+    <iframe 
+        src="https://demo.arcade.software/zBBJiIaTBpoBWErwvpbd?embed&show_copy_link=true"
```
[the remark this answers is not in the document]
Comment (PoojaB26) [reply docs/ff-concepts/adding-customization/custom-widgets.md:37]: fixed

File: docs/ff-concepts/adding-customization/custom-code.md [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+
+:::tip[What are callbacks?]
+
+In programming, callbacks are functions passed to other functions to be called when a specific event
+occurs.
+:::
+In the following example, we have a Custom Action that takes an `onError(searchKeyword)` callback 
+action with an Action Parameter `searchKeyword`. This means that the custom action will provide this search keyword back to the callback action when it calls it. 
+
+![explain-callback-action.png](imgs%2Fexplain-callback-action.png)
```
[the remark this answers is not in the document]
Comment (PoojaB26) [reply docs/ff-concepts/adding-customization/custom-code.md:166]: thanks, fixed

File: docs/ff-concepts/adding-customization/custom-code.md [hunk 4]
```diff
@@ -0,0 +1,289 @@
...
+    </iframe>
+</div>
+
+### Compile Code
+
+When you are done adding your code snippets, you can compile it to ensure there are no
+compilation errors. To do so, click the **Compile Code** button.
+
+
+<figure>
```
[the remark this answers is not in the document]
Comment (PoojaB26) [reply docs/ff-concepts/adding-customization/custom-code.md:85]: fixed

File: docs/ff-concepts/adding-customization/_category_.json [hunk 5]
```diff
@@ -0,0 +1,4 @@
+{
+  "label": "Adding Customization",
```
[the remark this answers is not in the document]
Comment (PoojaB26) [reply docs/ff-concepts/adding-customization/_category_.json]: Keeping it "Customizations" instead.

**Comment by PoojaB26** [comment]
@pinkeshmars comments addressed!

**Review by pinkeshmars — APPROVED** [review]

**Pull request merged**